soroban-env-host/src/test/auth.rs

```diff
@@ -86,6 +87,9 @@ impl SignNode {
 impl AuthTest {
     fn setup(signer_cnt: usize, contract_cnt: usize) -> Self {
         let host = Host::test_host_with_recording_footprint();
+        // FIXME: remove the `reset_unlimited` and instead reset inputs wherever appropriate
+        // to respect the budget limit.
+        host.as_budget().reset_unlimited();
```

Comment:
Why don't we do this generally in `pop_frame`? Also, what is the difficulty of using the proper reset instead of unlimited?

Reply:
I haven't understood those tests well enough to determine where are the boundaries to put in budget resets. 
Reset should happen in between different contract invocations, as well as between the preflight and actual run. 
If a contract invokes other contracts, the budget needs to be maintained so that all the costs need to be charged to the top level contract invocation. (which is why we can't reset budgets in `pop_frame`)